nixos/modules/system/boot/loader/systemd-boot/systemd-boot-builder.py

```diff
@@ -77,36 +88,45 @@ def mkdir_p(path):
         if e.errno != errno.EEXIST or not os.path.isdir(path):
             raise
 
-def get_generations(profile):
+def get_generations(profile=None):
     gen_list = subprocess.check_output([
         "@nix@/bin/nix-env",
         "--list-generations",
         "-p",
-        "/nix/var/nix/profiles/%s" % (profile),
+        "/nix/var/nix/profiles/%s" % ("system-profiles/" + profile if profile else "system"),
         "--option", "build-users-group", ""],
         universal_newlines=True)
     gen_lines = gen_list.split('\n')
     gen_lines.pop()
-    return [ int(line.split()[0]) for line in gen_lines ]
+    return [ (profile, int(line.split()[0])) for line in gen_lines ]
 
 def remove_old_entries(gens):
-    slice_start = len("@efiSysMountPoint@/loader/entries/nixos-generation-")
-    slice_end = -1 * len(".conf")
+    rex_profile = re.compile("^@efiSysMountPoint@/loader/entries/nixos-(.*)-generation-.*\.conf$")
```

Comment:
This is with intent. The `rex_profile` regex is used to decide it it is a system profile or not.